tools/offline_log_viewer/viewer.py

```diff
@@ -53,14 +54,35 @@ def print_groups(store):
     logger.info("")
 
 
+def validate_path(path):
+    if not os.path.exists(path):
+        logger.error(f"Path doesn't exist {path}")
+        sys.exit(1)
+    controller = join(path, "redpanda", "controller")
+    if not os.path.exists(controller):
+        logger.error(
+            f"Each redpanda data dir should have controller piece but {controller} isn't found"
+        )
+        sys.exit(1)
+
+
+def validate_topic(path, topic):
+    if topic:
+        topic = join(path, "kafka", topic)
+        if not os.path.exists(topic):
+            logger.error(f"Topic {topic} doesn't exist")
+            sys.exit(1)
+
+
 def main():
     import argparse
 
     def generate_options():
         parser = argparse.ArgumentParser(description='Redpanda log analyzer')
-        parser.add_argument('--path',
-                            type=str,
-                            help='Path to the log desired to be analyzed')
+        parser.add_argument(
+            '--path',
+            type=str,
```

Comment:
if you'd like to integrate more tightly with argument parsing, you can change `type = str` to `type = validate_path` and `validate_path` to:

```python
def validate_path(path):
    if not os.path.exists(path):
        raise ArgumentTypeError(f"Path doesn't exist {path}")
    controller = join(path, "redpanda", "controller")
    if not os.path.exists(controller):
        raise ArgumentTypeError(f"Each redpanda data dir should have controller piece but {controller} isn't found")
    return path
```